### PR TITLE
chore(github-action): Refine GitHub Actions criteria for builds, pull requests and release

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder <niladri.halder@mayadata.io>

The build.yml and pull_request.yml workflows will ignore pushes made to `changelogs/` directory, all .md files, LICENSE and MAINTAINERS files.

Added criteria to ignore build workflows on release.

### CODE REVIEW

**Comment by niladrih** [comment]
Related PRs:
https://github.com/openebs/node-disk-manager/pull/562
https://github.com/openebs/cstor-operators/pull/272
https://github.com/openebs/dynamic-localpv-provisioner/pull/41
https://github.com/openebs/zfs-localpv/pull/306
https://github.com/openebs/lvm-localpv/pull/42
https://github.com/openebs/jiva-operator/pull/66
https://github.com/openebs/maya/pull/1797
https://github.com/openebs/cstor-csi/pull/151
https://github.com/openebs/jiva/pull/350
https://github.com/openebs/velero-plugin/pull/151
https://github.com/openebs/m-exporter/pull/14
https://github.com/openebs/upgrade/pull/104
https://github.com/openebs/istgt/pull/353

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,9 +14,9 @@
 name: release
 
 on:
-  create:
-    tags:
-      - 'v*'
+  release:
+    types:
+      - 'created'
```
Comment (akhilerm) [inline .github/workflows/release.yml:22]: Tags need to be mentioned here.

**Review by akhilerm — CHANGES_REQUESTED** [review]
Some comments in the release workflow. Also can you make the same changes in all the related PRs. @niladrih

**Review by akhilerm — APPROVED** [review]
LGTM

**Comment by niladrih** [comment]
@akhilerm @shubham14bajpai -- Should a workflow job be added to `pull_request.yml` to build the cstor-pool image? Presently only cstor-base is being built.

**Comment by niladrih** [comment]
Leaving cstor-pool build out of pull_request.yml because cstor-base and cstor-pool just have a different entrypoint, and cstor-base build is sufficient.
cc: @shubham14bajpai

**Comment by akhilerm** [comment]
@shubham14bajpai PTAL.

**Review by shubham14bajpai — APPROVED** [review]
LGTM

**Pull request merged**